rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java

```diff
@@ -1560,6 +1561,73 @@ public void getFeePerKbElection_withVotes() {
         assertThat(result.getWinner(), is(expectedWinner));
     }
 
+    @Test
+    public void setLockingCap_before_fork() {
+        Repository repository = mock(Repository.class);
+        BridgeStorageConfiguration bridgeStorageConfiguration = new BridgeStorageConfiguration(false, false, false);
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                repository, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), bridgeStorageConfiguration
+        );
+
+        provider0.setLockingCap(Coin.ZERO);
+        provider0.saveLockingCap();
+
+        // If the network upgrade is not enabled we shouldn't be writing in the repository
+        verify(repository, never()).addStorageBytes(any(), any(), any());
+    }
+
+    @Test
+    public void setLockingCap_after_fork() {
+        Repository repository = mock(Repository.class);
+        BridgeStorageConfiguration bridgeStorageConfiguration = new BridgeStorageConfiguration(false, false, true);
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                repository, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), bridgeStorageConfiguration
+        );
+
+        provider0.setLockingCap(Coin.ZERO);
+        provider0.saveLockingCap();
+
+        // Once the network upgrade is active, we will store the locking cap in the repository
+        verify(repository, times(1)).addStorageBytes(
+                PrecompiledContracts.BRIDGE_ADDR,
+                DataWord.fromString("lockingCap"),
+                BridgeSerializationUtils.serializeCoin(Coin.ZERO)
+        );
+    }
+
+    @Test
+    public void setLockingCapAndGetLockingCap() {
+        Repository repository = createRepository();
+        Repository track = repository.startTracking();
+        BridgeStorageConfiguration bridgeStorageConfiguration = new BridgeStorageConfiguration(false, false, true);
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                track, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), bridgeStorageConfiguration
+        );
+
+        Coin expectedCoin = Coin.valueOf(666);
+
+        // We store the locking cap
+        provider0.setLockingCap(expectedCoin);
+        provider0.saveLockingCap();
+        track.commit();
+
+        track = repository.startTracking();
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+                track, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), bridgeStorageConfiguration
+        );
+
+        // And then we get it back
+        assertThat(provider.getLockingCap(), is(expectedCoin));
+    }
+
```

Comment:
Maybe you could add a test for getLockingCap after and before network upgrade activation without calling setLockingCap. I was lookging for a test where getInitialLockingCap was used